### ISSUE TITLE
ndarray::iter::Iter doesn't implement Debug

### ISSUE THREAD

**Issue by liam-clink** [rust-ndarray/ndarray#1204]
The title says it. What I was trying to do when I discovered this was print a zip of a range and an ndarray.

Like so:
```
let foo = ndarray::Array1::<f64>::zeros(100);
println!("{:?}", std::iter::zip(0..foo.len(), &foo);
```

**Comment by nilgoyette** [comment]
I never thought of printing an iterator before! Anyway, as per the official documentation:

> [fmt::Debug](https://doc.rust-lang.org/std/fmt/trait.Debug.html) implementations should be implemented for all public types.

So, yeah, I guess we should do it.

**Comment by biskwikman** [comment]
I'm not very experienced with Rust projects, but I'd like to try to implement this. Is Iter found in the iterators/mod.rs file?

**Comment by nilgoyette** [comment]
Yes, probably. Just clone the project and test, you will see.

**Comment by bluss** [comment]
Fixed in #1353, thanks and thanks @nilgoyette for mentoring
